### PR TITLE
Feature: groups in Gitea

### PR DESCRIPTION
**Pull request by techknowlogick**

### CODE REVIEW

**Comment by techknowlogick** [comment]
gentle ping for reviews.

**Comment by techknowlogick** [comment]
Conflicts have been resolved.

**Comment by sagikazarmark** [comment]
Thanks @techknowlogick !

I added reviewing it to my backlog: https://github.com/users/sagikazarmark/projects/1#card-53059357

**Comment by techknowlogick** [comment]
I've just resolved the conflicts.

**Comment by techknowlogick** [comment]
@sagikazarmark 👋 Hey, is there anything I can do to help get this merged?

**Comment by techknowlogick** [comment]
@nabokihms @sagikazarmark 👋 Hi, hope all is well. Is there anything I can do to help get this PR merged? Could I volunteer as a maintainer to take some load off your shoulders?

File: connector/gitea/gitea.go [hunk 1]
```diff
@@ -264,3 +417,10 @@ func (c *giteaConnector) user(ctx context.Context, client *http.Client) (giteaUs
 	}
 	return u, nil
 }
+
+// groupsRequired returns whether dex requires GitHub's 'read:org' scope. Dex
```
Comment (nabokihms) [inline connector/gitea/gitea.go:421]: I think the comment was just copied.
```suggestion
// groupsRequired returns whether dex requires Gitea 'read:org' scope. Dex
```

Comment (nabokihms) [reply connector/gitea/gitea.go:421]: And what about the other part of the comment? Especially the part about `org`. Is it correct?

File: connector/gitea/gitea.go [hunk 2]
```diff
@@ -232,9 +262,132 @@ func (c *giteaConnector) Refresh(ctx context.Context, s connector.Scopes, ident
...
+		for _, t := range teams {
+			groups = append(groups, t.Organization.Name)
+			groups = append(groups, formatTeamName(t.Organization.Name, t.Name))
+		}
+
+		page++
+	}
+
+	// remove dupelicate slice variables
+	keys := make(map[string]bool)
```
Comment (nabokihms) [inline connector/gitea/gitea.go:379]: NIT: We usually use map[string]struct{}

**Review by nabokihms — CHANGES_REQUESTED** [review]
Overall it looks good. Could you please write a test for the new feature? In addition, pay attention to group filters (maybe there is a way to reuse code).

**Comment by techknowlogick** [comment]
Thanks @nabokihms I've just updated the code comment, and coding style to align with existing dex code

**Comment by techknowlogick** [comment]
I'm not sure these CI fails are related, looking at them it is something to do with storage which this PR doesn't touch.

**Comment by sagikazarmark** [comment]
@techknowlogick 

```
Error: connector/gitea/gitea.go:355:17: undefined: ioutil
# github.com/dexidp/dex/connector/gitea [github.com/dexidp/dex/connector/gitea.test]
Error: connector/gitea/gitea.go:355:17: undefined: ioutil
FAIL	github.com/dexidp/dex/cmd/dex [build failed]
```

**Comment by techknowlogick** [comment]
@sagikazarmark ah thanks for catching that. I've resolved it, along with now passing the linter.

**Review by nabokihms — APPROVED** [review]
I am ok with this being merged.
1. There are some NITs left. Could you please look through them and apply suggestions?
2. We refactored out Docker build a little. Could you please rebase this PR on master in order to run the docker build job successfully?

**Comment by nabokihms** [comment]
@techknowlogick only DCO thing left.

**Comment by techknowlogick** [comment]
@nabokihms there we go, all passing. thanks for your help on this :)

**Comment by nabokihms** [comment]
@techknowlogick looks good to me. Thank you for your contribution!

**Pull request merged**